Amendment related to https://github.com/w3f/Grant-Milestone-Delivery/pull/1175

AltiMario opened the pull request:
#### Project Abstract

Develop a Tuxedo Web Wallet and Dapp.

Our goal is to showcase the potential of Polkadot using the UTXO paradigm for the end user, leveraging Tuxedo. The project will consist of two key stages:
- Stage 1: Developing a simple Web Wallet
- Stage 2: Developing a basic DApp

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ Polkadot (USDC & USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

AltiMario reviewed (commented):
Related to the evaluation https://github.com/w3f/Grant-Milestone-Delivery/pull/1175

takahser reviewed (changes requested):
@AltiMario is there any specific reason for skipping the docker deliverable?

PieWol commented:
Hey @AltiMario , I'd appreciate it if you could provide me some answers to the comments I raised. Thanks.

PieWol commented:
I'm expressing some urgency, as this PR is a prerequisite to continuing the milestone evaluation. @AltiMario

AltiMario commented:
sure thing @PieWol I will write it in a few minutes. Thanks

AltiMario commented:
Hi @PieWol sorry for the delay in my reply.
I think I covered all queries, for further doubts please don't hesitate to contact me.

1- "Is it correct that you have, instead of creating your own web-based wallet, integrated Talisman to be usable for Tuxedo and your dApp?"

Yes. We pivoted this part of the project following the suggestion of Joshua Orndorff. I added some extra info in the "Additional Information" section of milestone 1.

2- "Please adjust some links of this milestone to better fit the original application and its specifications for the deliverables. Right now the wallet readme isn't even mentioned in the documentation deliverable. Even though I think this is the key documentation for this milestone."

I completely updated the README file to better describe the changes done, but I forgot to add it to the file of milestone 1. I'm going to add the link.

3- "The demo for showcasing the Talisman wallet compatibility seems cut short. In the readme only a 4 second clip is visible."
Fixed.

4- "We will provide [...] a basic tutorial that explains how a user can use the web wallet app for Tuxedo (Private key configuration, local URI to be used to connect to blockchain) and send test transactions, tuxedo balance, which will show how the new functionality works.
Is this all covered by the readme within the wallet directory?"

Because it is not needed a tutorial of how to use Talisman we haven't done it. However, we created a basic video tutorial (3 parts) on how to use the extended CLI-wallet.

5- "Was listed as the specification for this deliverable. Yet in the milestone delivery you are linking to a little showcase video that walks through an offer creation and buying process. No info on private key configuration or how information from the chain is retrieved. It's just a video of a frontend usage." 

The Talisman integration doesn't cover the private key configuration but the extended CLI-wallet does.

6- "Is there no written article at all?", "Especially interesting could be a written article that covers your changes to Talisman to be Tuxedo compatible on a high level."

Sorry, I didn't link them properly. Now you can see it in the README and in milestone 1 (section documentation and article).  We produced documentation for: APIs spreadsheet, Build and run the project, DApp specification, Frontend development libs, Game design, Talisman wallet, Wireframes, Comparative analysis "CryptoKitties Development: A comparative analysis between EVM - Cardano - Tuxedo/Polkadot"

7- "Furthermore what exactly are you referring to by "webservice"? Is this supposed to be your dApp?"
The solution/architecture is described in the doc "DApp-specification" https://github.com/mlabs-haskell/TuxedoDapp/wiki/DApp-specification documentation. The dockerization for the Talisman integration is not needed. The extended CLI-wallet is an extension of the Tuxedo framework, it can be run in the same way. What is new is the webservice part, where the docker container is provided.

PieWol commented:
@AltiMario ,
got it. Thanks!

Finally, what is the work you did to make Talisman work with Tuxedo? I thought Tuxedo / UTXO wasn't supported. Is there any code you had to modify on Talismans side or are you somehow making this work with blind signing? If you can please provide me a link to the code.

AltiMario commented:
hi @PieWol we used the blind signing, no work was done on the Talisman codebase.
The signing (redeemer creation) for the webservice part is here https://github.com/mlabs-haskell/Tuxedo/blob/cf02ebc8cba89d398e4fc1016ee76b17d764bcd4/webservice-wallet/src/kitty.rs#L937
and for the frontend part is here https://github.com/mlabs-haskell/TuxedoDapp/blob/6f9def9ee2f654762f433c8605b5ba4132c272a2/src/api/utils.ts#L89
For further technical questions related to the core, cli-wallet and the webservice part I invite @NadigerAmit to answer you and @philoniare can address the frontend part.
Both developers are already on other projects but they are available to support the technical conversation in detail, explaining what they did.

PieWol commented:
Hey @AltiMario ,
thanks again for your feedback. Would you also please answer the comments I made in my review of this very PR, additionally to the points I raised in my evaluation? You should find these in this thread if you scroll up.

PieWol commented:
@AltiMario ,
With you pivoting from creating the web-wallet to simply using Talisman blind signing this seems to be less work than previously outlined. Thus you need to lower the cost of the first milestone retroactively. Would you please come up with a reasoned proposal for this? Thanks!

AltiMario commented:
hi @PieWol about your comment https://github.com/w3f/Grants-Program/pull/2305#issuecomment-2121885364 I don't know what I missed in this reply https://github.com/w3f/Grants-Program/pull/2305#issuecomment-2117183560
Could you please rephrase your concern or point out what hasn't been properly explained that deserves more clarification?

AltiMario commented:
@PieWol I think there is a misunderstanding of the PR amendment scope. It is not for reduced work or just for the Talisman integration. It is also for the CLI-wallet extension, which took @NadigerAmit almost 2 months to build. It is a lot of work, probably a bit hidden (?), but that was a mandatory thing to do in order to have the DApp working. We exposed the functions built for the CLI-wallet as APIs of the webservice. 
In this project, saving money was not on the agenda. In fact, the resources employed were twice what I initially planned (you can see it in the code contributors list), leading to a significant budget overrun. For MLabs this represented an investment in this technology/framework, with the goal (and hope) to bring further developments and work. 
For every question related to the huge amount of work done for the extended CLI-wallet please refer to @NadigerAmit who will be happy and proud to answer and drive you on his code.
Thanks, Mario

PieWol reviewed (approved):
Hey @AltiMario ,
please excuse my misunderstanding. I was under the false impression that the cli-wallet extension was already part of the application previous to the amendment. This amendment looks good to me. Thanks for your swift replies and I'm sorry for the inconvenience.

AltiMario commented:
Hi @PieWol, don't worry, it didn't cause any inconvenience. I'm glad the work has been approved. We have a common goal, to create value for this ecosystem.
Thanks to you and the rest of your team for the support.

NadigerAmit commented on applications/TuxedoDapp.md:
```diff
@@ -6,160 +6,158 @@
-- TypeScript 
-- Babel for backward compatibility in older Browsers 
-- IndexedDB for Local database
-- IPFS for storing metadata related to the kitty such as Name, Trade status, Price, etc.
```
@PieWol, Metadata of kitties is stored in the chain and it will not overload the chain. This was discussed with Joshy also and he agreed to our approach.
